plugins/CoreHome/angularjs/siteselector/siteselector-model.service.js

```diff
@@ -63,17 +64,31 @@
             }
 
             if (model.isLoading) {
-                model.currentRequest.abort();
+                if (model.currentRequest) {
+                    model.currentRequest.abort();
+                } else if (limitPromise) {
+                    limitPromise.abort();
+                    limitPromise = null;
+                }
             }
 
             model.isLoading = true;
 
-            model.currentRequest = piwikApi.fetch({
-                method: 'SitesManager.getPatternMatchSites',
-                pattern: term
-            });
+            if (!limitPromise) {
+                limitPromise = piwikApi.fetch({method: 'SitesManager.getNumWebsitesToDisplayPerPage'});
+            }
```

Comment:
Well, what I meant was adding it as a recognized attribute in the directive (ie, in the `scope:` property) not so much a twig template :)

It's probably not necessary to move it for 2.15, maybe it would be a good idea for 3.0, just to keep technical debt from collecting. I'll leave it up to you whether a new ticket should be created.